invisible_cities/io/mcinfo_io.py

```diff
@@ -172,13 +181,97 @@ def __call__(self, mctables: (tb.Table, tb.Table, tb.Table, tb.Table),
         self.generator_table.flush()
 
 
-def copy_mc_info(h5in : tb.File, writer : Type[mc_info_writer], which_events : List[int]=None):
-    """Copy from the input file to the output file the MC info of certain events.
+def mc_writer(h5out : tb.file.File) -> Callable:
+    mcwriter_ = partial(df_writer         ,
+                        h5out      = h5out,
+                        group_name =  'MC')
+    def write_mctables(table_dict : Dict):
+        first_indx = check_last_merge_index(h5out) + 1
+        for key, tbl in table_dict.items():
+            if (key is MCTableType.configuration or
+                key is MCTableType.event_mapping   ):
+                try:
+                    orig_indx = tbl.file_index.unique()
+                    new_indx  = np.arange(first_indx                 ,
+                                          first_indx + len(orig_indx))
+                    tbl.file_index.replace(orig_indx, new_indx, inplace=True)
+                except AttributeError:
+                    tbl['file_index'] = first_indx
+            mcwriter_(df=tbl, table_name=key.name)
+    return write_mctables
+
+
+def check_last_merge_index(h5out : tb.file.File) -> int:
+    """
+    Get the last file index used to index merged files
+    """
+    if 'MC' in h5out.root:
+        if 'event_mapping' in h5out.root.MC:
+            return h5out.root.MC.event_mapping.cols.file_index[-1]
+    return 0
+
+
+def read_mc_tables(file_in : str                 ,
+                   evt_arr : np.ndarray          ,
+                   db_file : Optional[str] = None,
+                   run_no  : Optional[int] = None) -> Dict:
+    """
+    Reads all the MC tables present in
+    file_in and sorts the Dataframes for
+    each into a dictionary.
+    """
+    mctbls = get_mc_tbl_list(file_in)
+    if (len(evt_arr) > 0 and
+        len(np.intersect1d(evt_arr, get_event_numbers_in_file(file_in))) == 0):
```

Comment:
This was to reproduce the behaviour of that was present from PR #702 and copy the tables without event number as a variable in case of 0 events passing the cuts in the city being launched